mmv1/products/cloudrun/api.yaml

```diff
@@ -616,6 +616,68 @@ objects:
                     required: true
                     description: |-
                       This must match the Name of a Volume.
+            - !ruby/object:Api::Type::NestedObject
+              name: startupProbe
+              min_version: beta
+              description: |-
+                Startup probe of application within the container.
+                All other probes are disabled if a startup probe is provided, until it
+                succeeds. Container will not be added to service endpoints if the probe fails.
+              properties:
+                - !ruby/object:Api::Type::Integer
+                  name: initialDelaySeconds 
+                  description: |-
+                    Number of seconds after the container has started before the probe is
+                    initiated.
+                    Defaults to 0 seconds. Minimum value is 0. Maximum value is 240.
+                - !ruby/object:Api::Type::Integer
+                  name: timeoutSeconds 
+                  description: |-
+                    Number of seconds after which the probe times out.
+                    Defaults to 1 second. Minimum value is 1. Maximum value is 3600.
+                    Must be smaller than periodSeconds.
+                - !ruby/object:Api::Type::Integer
+                  name: periodSeconds 
+                  description: |-
+                    How often (in seconds) to perform the probe.
+                    Default to 10 seconds. Minimum value is 1. Maximum value is 240.
+                - !ruby/object:Api::Type::Integer
+                  name: failureThreshold  
+                  description: |-
+                    Minimum consecutive failures for the probe to be considered failed after
+                    having succeeded. Defaults to 3. Minimum value is 1.
+                - !ruby/object:Api::Type::NestedObject
+                  name: tcpSocket 
+                  description: |-
+                    TcpSocket speficies an action involving a TCP port.
+                  properties:
+                    - !ruby/object:Api::Type::Integer
+                      name: port
```

Comment:
Should this field be made required? Otherwise, it will be possible to specify the following, likely resulting in an error:

```
tcp_socket {}
```

Reply:
```
tcp_socket {}
```
is a valid input.

Reply:
In our API, port is optional. Default value will be set for port if it is not specified.

However I tested with `tcp_socket {}` and got an error saying either tcp_scoket or http_get should be set. Does terraform ignore tcp_socket field if it is empty?

Reply:
If a user doesn't specify `tcp_socket`, the API won't return it.

The default value is equal to `container.ports[0].port`, which is input by user or defaults to `8080`.